src/utils/httpVariableProviders/environmentVariableProvider.ts

```diff
@@ -44,11 +53,25 @@ export class EnvironmentVariableProvider implements HttpVariableProvider {
         return Object.keys(variables).map(key => ({ name: key, value: variables[key]}));
     }
 
+    private getShared(name: string): HttpVariable {
+        const variables = this.getSharedVariables();
+        if (!(name in variables)) {
+            return { name, error: ResolveErrorMessage.EnvironmentReferenceVariableNotExist };
+        }
+        return { name, value: variables[name] };
+    }
+
     private async getAvailableVariables(): Promise<{ [key: string]: string }> {
```

Comment:
I suggest that `getAvailableVaraibles` returns the resolved variables. And for non-exist variables, no error is emitted, just leave it as is. And in this case, `getShared` and `getSharedVariables` are no longer needed.